### ISSUE TITLE
Potential bug: presence of a subpackage lets editable install point to 'project root dir' instead of 'src'

### ISSUE THREAD

**Issue by maichmueller** [scikit-build/scikit-build#579]
Hi all,

I have noticed a strange behaviour while building a package of mine, which may be due to my configuration or a bug.
I have compiled a test project with a blank python module that I am compiling using scikit-build. This project follows the 'src' layout, which comes down to this:
```
├── CMakeLists.txt
├── pyproject.toml
├── setup.py
└── src
    └── pkg
        ├── core
        │   └── __init__.py
        ├── __init__.py
        ├── module.cpp
        └── some_funcs.py

```

The weird part is this: The mere presence of the subpackage `core` lets `pip install -e .` point to the root directory of this project `.` instead of `src`, the actual package directory:
> Creating /home/michael/anaconda3/envs/testpkg/lib/python3.8/site-packages/pkg.egg-link (link to .)

 This causes the editable install to be useless, as one would need to call `import src.pkg` in python to import this. 
It points correctly to `src` when `core` is removed from the package, or when one does not use scikit-build (i.e. uses `setuptools.setup` and does not compile the extension:
> Creating /home/michael/anaconda3/envs/testpkg/lib/python3.8/site-packages/pkg.egg-link (link to src)

In order to reproduce this behaviour I have put together a minimal project that does this (https://github.com/maichmueller/scikit_build_issue). The steps to reproduce are:

```
git clone https://github.com/maichmueller/scikit_build_issue.git
cd scikit_build_issue
pip install -e .
```

In order to see it pointing to `src`, either do:
1) delete `core` from `src/pkg` or
2) uncomment the last lines in setup.py to use setuptools' `setup` function (and comment out skbuild's `setup` call).

Is this an error on my side due to faulty configuration? The non working setup py reads the following setup call:
```
setup(
    name="pkg",
    version="0.0.1",
    packages=find_packages(where="src"),
    package_dir={"": "src"},
    cmake_install_dir="src/pkg",
    cmake_args=[f"-Dpybind11_SEARCH_PATH={pybind11_path}"],
    zip_safe=False,
    include_package_data=True,
)
```

Help or feedback to what might be happening here would be appreciated! 

Thanks,
Michael

**Comment by maichmueller** [comment]
Just noticed my issue is a duplicate of #546

**Comment by maichmueller** [comment]
So I have been digging a little into the setuptools process and noticed where the editable install breaks with skbuild:

In `setuptools/command/egg_info.py` the egg-link (which needs to point to `src` in the end for the link to work) is determined incorrectly. Lines 220 - 222 set this link via:
```
        if self.egg_base is None:
            dirs = self.distribution.package_dir
            self.egg_base = (dirs or {}).get('', os.curdir)
```

the package_dir dictionary is set in the setup call:
```
setup(
    ...
    package_dir={"": "src"},
    ...
)
```
and is used as such in the normal setup process (without scikit-build). But for my repo case above, skbuild sets package_dir to:
```
{'pkg.core': 'src/pkg/core', 'pkg': '_skbuild/linux-x86_64-3.8/cmake-install/src/pkg'}
```
this causes the call to `.get('', os.curdir)` to evaluate to `os.curdir` which is simply `'.'`. Thats why the egg-link becomes the project dir in the end. 
Can you give any pointers into why the `package_dir` values are overwritten with the actual pkg directories, which are then passed to this simple handler by setuptools?

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
